components/script/dom/document.rs

```diff
@@ -4515,6 +4494,21 @@ impl Document {
         self.active_sandboxing_flag_set.set(flags)
     }
 
+    pub(crate) fn creation_sandboxing_flag_set(&self) -> SandboxingFlagSet {
+        self.creation_sandboxing_flag_set.get()
+    }
+
+    pub(crate) fn creation_sandboxing_flag_set_considering_parent_iframe(
+        &self,
+    ) -> SandboxingFlagSet {
+        self.window()
+            .window_proxy()
+            .frame_element()
+            .and_then(|element| element.downcast::<HTMLIFrameElement>())
+            .map(HTMLIFrameElement::sandboxing_flag_set)
```

Comment:
Should this also include:

> the flags set on embedder's [node document](https://dom.spec.whatwg.org/#concept-node-document)'s [active sandboxing flag set](https://html.spec.whatwg.org/multipage/browsers.html#active-sandboxing-flag-set)

per https://html.spec.whatwg.org/multipage/browsers.html#determining-the-creation-sandboxing-flags

It seems that the `unwrap_or_else` does this fallback, but I think we need to union the two.